python/cudf/cudf/tests/test_orc.py

```diff
@@ -724,3 +724,21 @@ def test_orc_bool_encode_fail():
     # Also validate data
     pdf = pa.orc.ORCFile(buffer).read().to_pandas()
     assert_eq(okay_df, pdf)
+
+
+def test_nanoseconds_overflow(tmpdir):
+    path = tmpdir.join("nano64bit.orc")
+    # Use nanosecond values that take more than 32 bits to encode
+    s = cudf.Series([710424008, -1338482640], dtype="datetime64[ns]")
+    expected = cudf.DataFrame({"s": s})
+    expected.to_orc(path)
+
+    cudf_got = cudf.read_orc(path)
+    assert_eq(expected, cudf_got)
+
+    try:
+        orcfile = pa.orc.ORCFile(path)
+    except pa.ArrowIOError as e:
+        pytest.skip(".orc file is not found: %s" % e)
+    pyarrow_got = orcfile.read()
+    assert_eq(expected.to_pandas(), pyarrow_got.to_pandas())
```

Comment:
```suggestion
def test_nanoseconds_overflow():
    buffer = BytesIO()
    # Use nanosecond values that take more than 32 bits to encode
    s = cudf.Series([710424008, -1338482640], dtype="datetime64[ns]")
    expected = cudf.DataFrame({"s": s})
    expected.to_orc(buffer)

    cudf_got = cudf.read_orc(buffer)
    assert_eq(expected, cudf_got)

    orcfile = pa.orc.ORCFile(buffer)
    pyarrow_got = orcfile.read()
    assert_eq(expected.to_pandas(), pyarrow_got.to_pandas())
```

We can use the buffer that way we can skip the file existence checks.

Reply:
I did not know that ORCFile supports buffer input, thanks!
I feel like this is a very common mistake in these tests.

Reply:
yeah, we've been switching them in any test we touch/add.